assets/js/components/ExecutionResults/ExecutionResults.test.jsx

```diff
@@ -511,4 +513,33 @@ describe('ExecutionResults', () => {
     userEvent.click(screen.getByText(description));
     expect(screen.queryByText(remediation)).not.toBeInTheDocument();
   });
+
+  it('should render PremiumPill if a check in catalog is premium', () => {
+    const {
+      clusterID,
+      clusterHosts,
+      loading,
+      catalog,
+      executionError,
+      executionStarted,
+      executionResult,
+      checks,
+    } = prepareStateData('completed');
+
+    renderWithRouter(
+      <ExecutionResults
+        clusterID={clusterID}
+        clusterHosts={clusterHosts}
+        catalogLoading={loading}
+        catalog={catalog}
+        executionLoading={loading}
+        executionStarted={executionStarted}
+        executionData={executionResult}
+        executionError={executionError}
+        clusterSelectedChecks={checks}
+      />
+    );
+    expect(screen.getByText('Premium')).toBeVisible();
```

Comment:
This test looks a bit weak. Could we maybe test that pill is put to the correct check row?

Reply:
Yeah fair point, i will make the test stronger!

Reply:
Thanks!
I think you don't need the second check on the `Premium` text (screen.getByTest('Premium'))`.
Once looks good enough hehe